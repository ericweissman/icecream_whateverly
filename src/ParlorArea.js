import React, { Component } from 'react';
import App from './App.js';
import { uid } from 'react-uid';
import './css/Main.scss';
import Card from './Card.js';
import ParlorCard from './ParlorCard.js';

class ParlorArea extends Component {
  constructor(props) {
    super(props)

  };

  render() {
    return ( 
      <div className="parlor-area">
        <div className="parlor-list">
          {
            this.props.parlors.filter((parlor) => {
              return parlor.flavors.includes(parseInt(this.props.id))
            })
            .map((parlor) => {
              return (
                <ParlorCard parlor={parlor}
                />
                )
              })
            } 
        </div>
            <button onClick={this.props.getMoreInfo} className="back-to-all-flavors">Back To All Flavors</button>
      </div>
      )
    };
  }
  

export default ParlorArea