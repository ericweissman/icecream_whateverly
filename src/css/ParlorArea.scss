.parlor-list {
  margin-top: 370px;
  display: grid;
  grid-template-columns: 1fr 1fr;
}