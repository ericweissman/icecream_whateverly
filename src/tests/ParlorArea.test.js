import React from 'react';
import ParlorArea from '../ParlorArea';
import { shallow } from 'enzyme';


let wrapper;

describe('ParlorArea', () => {

  beforeEach(() => {
    wrapper = shallow(
      <ParlorArea ParlorArea={ParlorArea} />
      )
    });
  })
  it('should match snapshot when all data is passed correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
