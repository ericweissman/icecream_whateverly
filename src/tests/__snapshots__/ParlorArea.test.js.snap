// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParlorArea should match snapshot when all data is passed correctly 1`] = `
<div
  className="parlor-area"
>
  <div
    className="parlor-list"
  >
    <ParlorCard
      parlor={
        Object {
          "address": "2620 16th St, Denver, CO 80211",
          "daysOpen": Array [
            Object {
              "close": "10 PM",
              "day": "Sunday",
              "open": "11 AM",
            },
            Object {
              "close": "10 PM",
              "day": "Monday",
              "open": "11 AM",
            },
            Object {
              "close": "10 PM",
              "day": "Tuesday",
              "open": "11 AM",
            },
            Object {
              "close": "10 PM",
              "day": "Wednesday",
              "open": "11 AM",
            },
            Object {
              "close": "10 PM",
              "day": "Thursday",
              "open": "11 AM",
            },
            Object {
              "close": "11 PM",
              "day": "Friday",
              "open": "11 AM",
            },
            Object {
              "close": "11 PM",
              "day": "Saturday",
              "open": "11 AM",
            },
          ],
          "description": "Quirky shop for exotic ice cream",
          "flavors": Array [
            1,
            2,
            3,
            4,
            5,
            6,
            7,
            8,
            9,
            10,
            11,
            12,
            13,
            14,
            15,
            16,
            17,
            18,
            19,
            20,
            21,
            22,
            23,
            24,
            25,
            26,
            27,
          ],
          "numberOfFlavors": 27,
          "parlorName": "Little Man Ice Cream",
          "phoneNumber": "303-455-3811",
          "priceRange": "$",
          "review": 4.7,
          "website": "http://www.littlemanicecream.com/latest-scoop/menu/",
        }
      }
    />
    <button
      className="back-to-all-flavors"
      onClick={[MockFunction]}
    >
      Back To All Flavors
    </button>
  </div>
</div>
`;

exports[`should match snapshot when all data is passed correctly 1`] = `undefined`;
